GLMakie/src/events.jl

```diff
@@ -170,11 +173,12 @@ end
 
 function correct_mouse(screen::Screen, w, h)
     nw = to_native(screen)
-    sf = screen.scalefactor[] / (@static Sys.isapple() ? scale_factor(nw) : 1)
     _, winh = window_size(nw)
-    @static if Sys.isapple()
+    if GLFW.GetPlatform() in (GLFW.PLATFORM_COCOA, GLFW.PLATFORM_WAYLAND)
+        sf = screen.scalefactor[] / scale_factor(nw)
         return w, (winh / sf) - h
     else
+        sf = screen.scalefactor[]
         return w / sf, (winh - h) / sf
     end
 end
```

Comment:
This isn't something that changed in this pr but the scaling with mac/wayland seems wrong to me. The GLFW docs state that GetCursorPos (`w, h`) and GetWindowSize (`window_size(nw)`) both return screen coordinates, so it should be `winh - h` not `winh / sf - h`. The scalefactor from GetContentScale (`scale_factor(nw)` should move that to pixel coordinates which I'm not sure we want here. Dropping the scaling of `w` here also seems odd.

Maybe you can check if
```julia
f, a, p = scatter(rand(10))
on(println, events(f).mouseposition)
display(f) # scalefactor = 2
```
gives you the same range of mouse coordinates as `f.scene.viewport[]` suggests within the window. For me on Windows this is always true, including with an OS level scale factor and a Makie level scale factor. If that's not the case you probably just want the else branch for Wayland here.

Reply:
I think the output indicates things are working as expected?
```

julia> display(f, scalefactor=2)
GLMakie.Screen(...)

julia> (0.9375, 0.3046875)
(0.9375, 0.84375)
(0.9375, 0.50390625)
(599.98828125, 449.3515625)
julia> f.scene.viewport[]
GeometryBasics.HyperRectangle{2, Int64}([0, 0], [600, 450])
```

The thing I tested when modifying these conditions was that click-and-dragging over a region to zoom in worked like expected, i.e. that I could select where I wanted to and the selection highlight followed my mouse.

But...I **do** have problems if I force the scalefactor to something other than the native 2 for my screen, so I think you're probably right that somethings not quite right. (I too was curious that `w` isn't scaled, but I was trying to limit my changes to just changing the platform-specific conditionals.)

Reply:
Yea I think if the scalefactor matches the OS you just have a 1.0 multiplier here so the order doesn't matter.

I don't know whether this was correct on mac. I'd like to have that checked before merging this pr, but I don't have one to check on myself...

Reply:
A quick search of the issues for OSX mouse problems lead me to #3429, which reads to me like confirmation that the previous behavior wasn't working correctly on OSX. Maybe someone from that thread (@jacobleft or @pbouffard) can confirm that this branch either fixes the problem or at minimum doesn't make the situation even worse.